### PR TITLE
Use make + size to eliminate n+1 grow container issue

### PR DESCRIPTION
**Pull request by edward-yakop**
For convert orders:
Before:
image

After:
image

For flatten:
Before;
image

After:
image

### CODE REVIEW

**Comment by nmoniz** [comment]
Hi, have you considered using `make(..., 0, size)` and keep using the `append`?

File: api.go [hunk 1]
```diff
@@ -3,7 +3,7 @@ package luno
 import (
 	"context"
 
-	"github.com/luno/luno-go/decimal"
```
Comment (obalunenko) [inline api.go:6]: Try not to change module names when creating PR's to public repos from your forks.

**Review by obalunenko — CHANGES_REQUESTED** [review]

File: api.go (hunk 1 above)
Comment (echarrod) [reply api.go:6]: You can use a replace directive in your go.mod file to be able to leave these lines unmodified, yet still use your local module. See [this SO answer](https://stackoverflow.com/questions/14323872/using-forked-package-import-in-go) for an example

**Comment by echarrod** [comment]
Thanks for the contribution! 

Will close this as it hasn't had any activity for a while. Feel free to open it again if you are able to make the updates based on the suggestions 👍

**Pull request closed without merging**